openedx_learning/apps/authoring/collections/api.py

```diff
@@ -69,6 +85,82 @@ def update_collection(
     return collection
 
 
+def add_to_collections(
+    collections_qset: QuerySet[Collection],
+    contents_qset: QuerySet[PublishableEntity],
+) -> int:
+    """
+    Adds a QuerySet of PublishableEntities to a QuerySet of Collections.
+
+    Records are created in bulk, and so integrity errors are deliberately ignored: they indicate that the content(s)
+    have already been added to the collection(s).
+
+    Returns the number of entities added (including any that already exist).
+    """
+    collection_objects = []
+    object_ids = contents_qset.values_list("pk", flat=True)
+
+    for collection in collections_qset.only("pk").all():
+        for object_id in object_ids:
+            collection_objects.append(
+                CollectionObject(
+                    collection_id=collection.pk,
+                    object_id=object_id,
+                )
+            )
+
+    created = CollectionObject.objects.bulk_create(
+        collection_objects,
+        ignore_conflicts=True,
+    )
+    return len(created)
+
+
+def remove_from_collections(
+    collections_qset: QuerySet,
+    contents_qset: QuerySet,
+) -> int:
+    """
+    Removes a QuerySet of PublishableEntities from a QuerySet of Collections.
+
+    PublishableEntities are deleted from each Collection, in bulk.
+
+    Returns the total number of entities deleted.
+    """
+    total_deleted = 0
+    object_ids = contents_qset.values_list("pk", flat=True)
+
+    for collection in collections_qset.only("pk").all():
+        num_deleted, _ = CollectionObject.objects.filter(
+            collection_id=collection.pk,
+            object_id__in=object_ids,
+        ).delete()
+
+        total_deleted += num_deleted
+
+    return total_deleted
+
+
+def get_collection_components(collection_id: int, component_qset: QuerySet[Component]) -> QuerySet[Component]:
+    """
+    Returns the Components associated with a given Collection.
+
+    Filters matching Components from the given components queryset.
+    """
+    coll_obj_ids = CollectionObject.objects.filter(collection_id=collection_id).values_list("object_id", flat=True)
+    return component_qset.filter(publishable_entity_id__in=coll_obj_ids).order_by("pk")
```

Comment:
Hmm.. I thought we'd need this for https://github.com/openedx/modular-learning/issues/230 too, but now I'm unsure.. might only need `get_object_collections` below, since we can search for a Collection's components with a filter for `collection='<id>'`.